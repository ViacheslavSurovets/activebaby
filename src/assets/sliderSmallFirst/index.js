import barcelona from './barcelona.jpg';
import barletta from './barletta.jpg';
import benita from './benita.jpg';
import chantal from './chantal.jpg';
import cristiano from './cristiano.jpg';
import giano from './giano.jpg';
import massimo from './massimo.png';
import sierra from './sierra.jpg';
import riccio from './riccio.jpg';
import neonex from './neonex.jpg';
import monte from './monte.png';

export {
  barcelona,
  barletta,
  benita,
  chantal,
  cristiano,
  giano,
  massimo,
  riccio,
  sierra,
  monte,
  neonex
};
