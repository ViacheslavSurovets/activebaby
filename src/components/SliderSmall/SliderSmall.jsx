import React, { useCallback } from 'react';
import Swiper from '../../core/swiper';
import { SliderSmallCard } from '@components/SliderSmallCard';
import PropTypes from 'prop-types';

const SliderSmall = ( { items } ) => {
  const params = {
    initialSlide: 3,
    ContainerEl: 'section',
    slidesPerView: window.innerWidth < 750 ? 3 : 5,
    navigation: {
      nextEl: '.swiper-button-next',
      prevEl: '.swiper-button-prev'
    },
    spaceBetween: 30,
    loop: true,
    autoplay: true
  };

  const renderItem = useCallback (
    ( { idx, url, title } ) => (
      <SliderSmallCard url={ url } title={ title } key={ `slide_${ idx }` } />
    ),
    []
  );

  return (
    <Swiper params={ params }>
      { items.map ( renderItem ) }
    </Swiper>
  );
};

export default SliderSmall;

SliderSmall.propTypes = {
  items: PropTypes.array.isRequired
};
