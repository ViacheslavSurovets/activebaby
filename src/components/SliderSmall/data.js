import { cristiano, chantal, adamex } from '@assets/sliderMain';

export default ( content = 'Slide' ) => {

  const dataSliderMain = [
    {
      url:cristiano,
      title: 'Cristiano',

    },
    {
      url:chantal,
      title: 'Chantal'
    },
    {
      url:adamex,
      title: 'Adamex'
    },
  ];


  return dataSliderMain.map ( ( item, idx ) => ({
    idx: idx + 1,
    content: `${ content } #${ idx + 1 }`,
    url : item.url,
    title: item.title,
    label : 'label'
  }) );
};
