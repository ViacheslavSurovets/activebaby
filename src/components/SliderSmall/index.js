export { default as SliderSmall } from './SliderSmall';