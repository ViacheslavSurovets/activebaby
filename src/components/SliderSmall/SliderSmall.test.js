import React from 'react';
import { shallow } from 'enzyme';
import SliderSmall from './SliderSmall';
import * as img from '@assets/sliderSmallFirst';


describe ( 'SliderSmall', () => {
  const mockItems = [
    {
      imageUrl: img.cristiano,
      title: 'Cristiano',
      path: '#',
      label: 'string'
    }
  ];

  it ( 'SearchMobile Snapshot', () => {
    expect ( shallow ( <SliderSmall items={ mockItems } /> ) ).toMatchSnapshot ();
  } );
} );
