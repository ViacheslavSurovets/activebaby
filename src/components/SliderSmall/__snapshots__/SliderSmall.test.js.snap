// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderSmall SearchMobile Snapshot 1`] = `ShallowWrapper {}`;
